test/unit/request_correlator_test.js

```diff
@@ -0,0 +1,184 @@
+    serverMocks = require('../tools/serverMocks'),
+    proxyLib = require('../../lib/fiware-pep-steelskin'),
+    orionPlugin = require('../../lib/plugins/orionPlugin'),
+    keystoneAuth = require('../../lib/services/keystoneAuth'),
+    proxyPlugin = require('../../lib/middleware/proxy'),
+    config = require('../../config'),
+    utils = require('../tools/utils'),
+    should = require('should'),
+    async = require('async'),
+    request = require('request'),
+    correlatorIds = [],
+    correlatorIdsPost = [],
+    sendRequestBackup;
+
+describe('Request correlators', function() {
+    var proxy,
+        mockTarget,
+        mockTargetApp,
+        mockAccess,
+        mockAccessApp,
+        mockOAuth,
+        mockOAuthApp,
+        currentAuthentication = {
+            module: 'keystone',
+            path: '/v3/role_assignments',
+            authPath: '/v3/auth/tokens',
+            rolesFile: './test/keystoneResponses/rolesOfUser.json',
+            authenticationResponse: './test/keystoneResponses/authorize.json',
+            headers: [
+            ],
+            authMock: serverMocks.mockKeystone
+        };
+
+    function initializeUseCase(currentAuthentication, done) {
+        config.authentication.module = currentAuthentication.module;
+        config.authentication.path = currentAuthentication.path;
+        config.authentication.authPath = currentAuthentication.authPath;
+
+        proxyLib.start(function(error, proxyObj) {
+            var testExtraction = function(req, res, callback) {
+                correlatorIds.push(domain.active.corr);
+                callback(null, req, res);
+            };
+
+            proxy = proxyObj;
+
+            proxy.middlewares.push(orionPlugin.extractCBAction);
+            proxy.middlewares.push(testExtraction);
+
+            serverMocks.start(config.resource.original.port, function(error, server, app) {
+                mockTarget = server;
+                mockTargetApp = app;
+                serverMocks.start(config.access.port, function(error, serverAccess, appAccess) {
+                    mockAccess = serverAccess;
+                    mockAccessApp = appAccess;
+                    serverMocks.start(config.authentication.options.port, function(error, serverAuth, appAuth) {
+                        mockOAuth = serverAuth;
+                        mockOAuthApp = appAuth;
+
+                        mockOAuthApp.handler = currentAuthentication.authMock;
+
+                        mockAccessApp.handler = function(req, res) {
+                            res.set('Content-Type', 'application/xml');
+                            res.send(utils.readExampleFile('./test/accessControlResponses/permitResponse.xml', true));
+                        };
+
+                        done();
+                    });
+                });
+            });
+        });
+    }
+
+    var options = {
+        uri: 'http://localhost:' + config.resource.proxy.port + '/NGSI10/updateContext',
```

Comment:
Yes. The test launches a local instance of the PEP Proxy in order to execute the test. This instance is then instrumented changing its middlewares and decorating some functions (as you can see in the last test) that will be impossible (or extremely difficult) to test remotely.